### PR TITLE
Adding password option to linux version of the formula

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: join-domain/elx/pbis/files/join.sh [hunk 1]
```diff
@@ -21,116 +21,275 @@
...
+  export SVCPASS
+
+  # Bail if needed decrypt failed
+  if [[ ${SVCPASS} == FAILURE ]]
+  then
+    printf "\n"
+    printf "changed=no comment='Failed to decrypt password'\n"
+    exit 1
+  else
+    printf "Decrypted password for sevice-account [%s]\n\n" "${SVCACCT}"
```
Comment (lorengordon) [inline join-domain/elx/pbis/files/join.sh]:
```suggestion
    printf "Decrypted password for service-account [%s]\n\n" "${SVCACCT}"
```

File: join-domain/elx/pbis/init.sls [hunk 2]
```diff
@@ -30,8 +30,13 @@ PBIS-NETBIOSfix:
...
 PBIS-join:
   cmd.script:
-    - name: 'join.sh "{{ join_domain.netbios_name }}" "{{ join_domain.dns_name }}" "{{ join_domain.username }}" "{{ join_domain.encrypted_password }}" "{{ join_domain.key }}" "{{ join_domain.oupath }}"'
+    {%- if join_domain.get("password") %}
+    - name: 'join.sh -n "{{ join_domain.netbios_name }}" -f "{{ join_domain.dns_name }}" -u "{{ join_domain.username }}" -p "{{ join_domain.password }}" -o "{{ join_domain.oupath }}"'
+    {%- else %}
+    - name: 'join.sh -n "{{ join_domain.netbios_name }}" -f "{{ join_domain.dns_name }}" -u "{{ join_domain.username }}" -c "{{ join_domain.encrypted_password }}" -k "{{ join_domain.key }}" -o "{{ join_domain.oupath }}"'
+    {%- endif %}
     - source: 'salt://{{ files }}/join.sh'
+    - template: jinja
```
Comment (lorengordon) [inline join-domain/elx/pbis/init.sls]: I don't think this script is now templated, since it wasn't before? Any reason this is being added?

Comment (confusdcodr) [reply join-domain/elx/pbis/init.sls]: Need to look at it again. I think I started to template pieces of it but then removed it and forgot about this line.

**Comment by confusdcodr** [comment]
Most of the ECLINT changes should be limited to the last commit so if you just look at the first 3 commits it may help out.

**Comment by confusdcodr** [comment]
> Most of the ECLINT changes should be limited to the last commit so if you just look at the first 3 commits it may help out.

File: join-domain/elx/pbis/files/join.sh (hunk 1 above)
Comment (confusdcodr) [reply join-domain/elx/pbis/files/join.sh]: Fixed

File: .travis.yml [hunk 3]
```diff
@@ -15,18 +15,24 @@ matrix:
         - OS_VERSION=6
         - SALT_PILLARROOT=$TRAVIS_BUILD_DIR/tests/pillar/test-linux-main
         - SALT_STATE=join-domain
+        - NODE_VERSION=10.x
```
Comment (lorengordon) [inline .travis.yml]: The way to do this is to setup a new `env`...

```
    - env:
        - TESTENV=eclint
        - ECLINT_OPTS="-s bash"
        - NODE_VERSION=10.x
        - NODE_SOURCE=https://deb.nodesource.com/setup_$NODE_VERSION
```

Then extend the if/elif blocks in the install and script sections.

Comment (confusdcodr) [reply .travis.yml]: Fixed

File: join-domain/elx/pbis/files/join.sh [hunk 4]
```diff
@@ -21,116 +21,275 @@
...
+fi
+
+# Decrypt our query password (as necessary)
+if [[ ${SVCPASS} == TOBESET ]]
+then
+  SVCPASS="$(PWdecrypt)"
+  export SVCPASS
+
+  # Bail if needed decrypt failed
+  if [[ ${SVCPASS} == FAILURE ]]
```
Comment (lorengordon) [inline join-domain/elx/pbis/files/join.sh]: `PWdecrypt` does not return FAILURE on error, it returns an empty string. The prior test was functioning...
```suggestion
  if [[ -z "${SVCPASS}" ]]
```

Comment (confusdcodr) [reply join-domain/elx/pbis/files/join.sh]: Copy pasta error on my part. Fixed.

**Comment by lorengordon** [comment]
🚢 let's give it a shot!

**Pull request merged**